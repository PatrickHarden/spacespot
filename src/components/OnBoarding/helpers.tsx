import {
  OnBoardingState,
  SpaceData,
  SpaceStatus,
  AvailabilityFlexPriceInterval,
} from 'services/onboarding/types'

const date = new Date('2020/01/01')

export const createSpace = (type: string): SpaceData => ({
  id: '0',
  type,
  spaceName: '',
  spaceDescription: '',
  spaceHighlights: '',
  spaceSize: undefined,
  spaceFloor: undefined,
  spaceSizeCommon: undefined,
  spaceMatterPort: '',
  spaceFloored: '',
  use: '',
  spaceRent: undefined,
  spaceServices: undefined,
  spaceServicesNotNegotiable: false,
  availabilityFixed: date,
  months: undefined,
  hotDesks: {
    desks: 0,
    availableFrom: date,
    minLease: 0,
    price: 0,
    currencyCode: 'NOK',
    frequency: AvailabilityFlexPriceInterval.Monthly,
  },
  fixedDesks: {
    desks: 0,
    availableFrom: date,
    minLease: 0,
    price: 0,
    currencyCode: 'NOK',
    frequency: AvailabilityFlexPriceInterval.Monthly,
  },
  servicedOffices: [],
  images: [],
  floorPlan: [],
  fitout: [{ description: 'FitOut1', name: 'FitOut1', amount: 1000 }],
  status: SpaceStatus.ReadyForPublish,
})

export const createOnBoardingState = (): OnBoardingState => ({
  isNew: true,
  buildingId: undefined,
  images: [
    {
      key: 0,
      name: '40VilliersStreet.External_Photo_1_small.png',
      type: 'image/png',
      uri:
        'data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAYAAAAfFcSJAAAADUlEQVR42mP8/5+hHgAHggJ/PchI7wAAAABJRU5ErkJggg==',
    },
  ],
  docs: [],
  name: '',
  description: '',
  address: '',
  city: '',
  postCode: '',
  country: '',
  location: '',
  latLng: undefined,
  district: '',
  subdistrict: {
    id: 0,
    name: '',
  },
  amenities: { hasGym: { checked: true, desc: 'Gym' } },
  spaces: { '0': createSpace('FLEX') },
  spaceCount: 1,
})

export const valInteger = (
  current: number | undefined,
  value: number,
  min: number,
  max?: number,
): number => {
  const val = Number(Number(value).toFixed())
  return val < min || (max && val > max) ? current || 0 : val
}
