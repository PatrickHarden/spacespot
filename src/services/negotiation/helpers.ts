import { get } from 'lodash'
import {
  NegotiationTerms,
  NegotiationTerm,
  NegotiationTermFields,
  NegotiationTermFieldsIds,
} from './types'

export const getNegotiationTermArray = (
  terms: NegotiationTerms,
  ids: NegotiationTermFieldsIds,
) => {
  const values: Array<NegotiationTerm> = Object.keys(terms).map(key => {
    return {
      termId: ids[key],
      value: get(terms, key),
    }
  })
  return values
}

const decapitalize = (title: string) => title[0].toLowerCase() + title.slice(1)

export const getFields = (
  fields: NegotiationTermFields,
  negTerms: Array<NegotiationTerm>,
): NegotiationTerms => {
  const initialTerms = {} as NegotiationTerms
  if (!negTerms || !fields) return initialTerms
  return (negTerms.reduce(
    (acc, term) => ({
      ...acc,
      [decapitalize(fields[term.termId])]: term.value,
    }),
    initialTerms,
  ) as unknown) as NegotiationTerms
}
