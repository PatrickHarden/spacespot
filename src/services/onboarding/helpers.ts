import { get, cloneDeep } from 'lodash'
import { Space } from 'services/space/types'
import spaceSelectors from 'services/space/selectors'
import {
  AmenityOptions,
  OnBoardingState,
  AvailabilityFlexPriceInterval,
  AvailabilityFlex,
} from 'services/onboarding/types'

export const aspects2Amenities = (
  amenities: AmenityOptions,
  aspects: Array<string>,
): AmenityOptions => {
  const newAmenities = cloneDeep(amenities) || {}
  if (!aspects) {
    return newAmenities
  }
  aspects.forEach(key => {
    if (newAmenities[key]) {
      newAmenities[key].checked = true
    } else {
      newAmenities[key] = {
        checked: true,
        desc: key,
      }
    }
  })
  return newAmenities
}

export const loadOnboardingSpace = (
  defaultAmenities: AmenityOptions | undefined,
  space: Space,
  isBuilding: boolean,
): OnBoardingState => {
  const building = spaceSelectors.translateBuilding(space, isBuilding)
  return {
    buildingId: building.buildingId,
    isNew: false,
    images: building.images || [],
    docs: [],
    name: building.name,
    description: building.description,
    address: building.address,
    city: building.city,
    postCode: building.postCode,
    country: building.country,
    location: building.location || '',
    latLng: building.latLng,
    district: building.district,
    subdistrict: building.subdistrict,
    amenities: aspects2Amenities(
      defaultAmenities || {},
      get(space, 'Common.Aspects', []),
    ),
    spaces: isBuilding ? {} : spaceSelectors.translateSpace(space),
    spaceCount: 0,
    uploadedImages: building.uploadedImages,
    uploadedDocs: building.uploadedDocs,
    isPublishing: false,
    defaultAmenities: defaultAmenities,
  }
}

const defaultDate = new Date()
defaultDate.setDate(new Date().getDate() + 1)

export const getDefaultAvailability = (): AvailabilityFlex => ({
  desks: 0,
  availableFrom: defaultDate,
  minLease: 0,
  price: 0,
  currencyCode: 'NOK',
  frequency: AvailabilityFlexPriceInterval.Monthly,
})
