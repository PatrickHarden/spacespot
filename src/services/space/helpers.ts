import { get, chain } from 'lodash'
import { Space } from './types'
import {
  CommonChargeType,
  CommonImage,
  ImageResource,
  CommonPostalAddress,
  CommonActualAddress,
} from 'services/space/PropertyListingschema'
import { FLEX_TYPE_CONSTANT, FIXED_TYPE_CONSTANT } from '../onboarding/utils'
import { routingLang } from 'intl'

import CONFIG from 'config'

export const generateGallery = (item: any) => {
  const [url] = item['Common.ImageResources'].filter(
    (photo: any) => photo['Common.Breakpoint'] === 'small',
  )
  const alt = item['Common.ImageCaption']
  return {
    url: url['Source.Uri'],
    alt,
  }
}

type Charges = {
  rent: number
  serviceCharges: number
  additionalCosts: number
  currencyCode: string
  total: number
}

export const getCharges = (space: Space) => {
  let charges: Charges = {
    rent: 0,
    serviceCharges: 0,
    additionalCosts: 0,
    currencyCode: 'EUR',
    total: 0,
  }
  const ccharges: Array<CommonChargeType> = get(space, 'Common.Charges', [])
  charges = ccharges.reduce((acc, charge) => {
    const kind = charge['Common.ChargeKind']
    const value = charge['Common.Amount'] || 0
    const currency = charge['Common.CurrencyCode']
    if (currency) {
      acc.currencyCode = currency
    }
    switch (kind) {
      case 'Rent':
        acc.rent = value
        break
      case 'ServiceCharge':
        acc.serviceCharges += value
        break
      default:
        acc.additionalCosts += value
    }
    acc.total += value
    return acc
  }, charges)
  return charges
}

export const getImages = (space: Space) => {
  const photos = get(space, 'Common.Photos')
  const urls = photos
    ? photos.map((item: CommonImage) => {
      const [original] = item['Common.ImageResources'].filter(
        (photo: ImageResource) => photo['Common.Breakpoint'] === 'original',
      )
      const [small] = item['Common.ImageResources'].filter(
        (photo: ImageResource) => photo['Common.Breakpoint'] === 'small',
      )
      return {
        original:
          original['Source.Uri'] ||
          CONFIG.GL_PHOTO_HOST + original['Common.Resource.Uri'],
        thumbnail:
          small['Source.Uri'] ||
          CONFIG.GL_PHOTO_HOST + small['Common.Resource.Uri'],
      }
    })
    : []
  return urls
}

export const getFloorPlans = (space: Space) => {
  const floorPlans = get(space, 'Common.FloorPlans')
  const urls = floorPlans
    ? floorPlans.map((item: CommonImage) => {
      const original = item['Common.ImageResources'][0]
      return {
        original:
          original['Source.Uri'] ||
          CONFIG.GL_PHOTO_HOST + original['Common.Resource.Uri'],
        thumbnail:
          original['Source.Uri'] ||
          CONFIG.GL_PHOTO_HOST + original['Common.Resource.Uri'],
      }
    })
    : []
  return urls
}

export const getPhoto = (space: Space): string => {
  const photo = chain(space)
    .get('Common.Photos')
    .head()
    .get('Common.ImageResources')
    .filter((img: ImageResource) => img['Common.Breakpoint'] === 'small')
    .head()
    .get('Source.Uri')
    .value()
  return photo || ''
}

export const getType = (space: Space) =>
  get(space, 'Common.UsageType') === 'Office' //FIXME when mapping changes
    ? FIXED_TYPE_CONSTANT
    : FLEX_TYPE_CONSTANT

export const isFixed = (space: Space) => getType(space) === FIXED_TYPE_CONSTANT

export const getSize = (space: Space) => {
  if (isFixed(space)) {
    return get(space, "['Common.MinimumSize'][0]['Common.Size']", 0)
  }
  return 0 // FIXME
}

/**
 * Size units (i.e: sqm)
 */
export const getSizeUnits = (space: Space) => {
  if (isFixed(space)) {
    return get(space, "['Common.MinimumSize'][0]['Common.Units']", 0)
  }
  return 0 // FIXME
}

/**
 * Calculate desks from size for fixed spaces
 *
 * desks = size in sqm / 15
 *
 * - For flex/coworking returns empty string
 * - If units != sqm or sqft, returns size with units
 *
 */
export const getDesks = (space: Space) => {
  if (isFixed(space)) {
    const sz: number = getSize(space)
    const units = getSizeUnits(space)
    if (units === 'sqm' || units === 'm2') {
      return (sz / 15).toFixed()
    } else if (units === 'sqft') {
      return ((sz * 0.092903) / 15).toFixed()
    } else {
      // unknown units
      return `${sz} ${units}`
    }
  }
  return ''
}

/**
 * Get common area (shared) size
 *
 * Shared common area = TotalSize - Size
 */
export const getSizeCommon = (space: Space) => {
  if (isFixed(space)) {
    return (
      get(space, "['Common.TotalSize'][0]['Common.Size']", 0) - getSize(space)
    )
  }
  return 0
}

export const getFloor = (space: Space) => get(space, 'Common.FloorNumber')

const getPostalAddress = (space: Space): CommonPostalAddress => {
  const actual: CommonActualAddress = get(space, 'Common.ActualAddress')
  const postal: Array<CommonPostalAddress> = get(
    actual,
    'Common.PostalAddresses',
    [],
  )
  return postal[0]
}

export const getLocation = (space: Space) =>
  get(space, '["Common.LocationDescription"][0]["Common.Text"]')

export const getTitle = (space: Space): string =>
  get(
    space,
    '["Common.Strapline"][0]["Common.Text"]',
    get(getPostalAddress(space), 'Common.Line1', ''),
  )

export const getStreet = (space: Space): string => {
  const address = getPostalAddress(space)
  return get(address, 'Common.Line2', get(address, 'Common.Line1', ''))
}

export const getDistrict = (space: Space) => {
  const address = getPostalAddress(space)
  return get(address, 'Common.Line3', '')
}

export const getSubdistrict = (space: Space) => {
  const address = getPostalAddress(space)
  return {
    id: 0,
    name: get(address, 'Common.Line4', ''),
  }
}

export const getCity = (space: Space): string => {
  const address = getPostalAddress(space)
  return get(address, 'Common.Locallity', '')
}

export const getPK = (space: Space): string =>
  get(space, 'Common.PrimaryKey', '')

export const getSpaceURI = (space: Space): string => {
  const office = routingLang({ nb: 'kontor', default: 'office' })
  const coworking = 'coworking'

  const id = getPK(space)
  const address = getStreet(space).replace(/\//g, '-')
  const district = (getDistrict(space) || 'NA').replace(/\//g, '-')
  const subdistrict = (getSubdistrict(space).name || 'NA').replace(/\//g, '-')
  const city = getCity(space)
  const title = getTitle(space).replace(/\//g, '-')
  const stype = isFixed(space) ? office : coworking
  const uri = `detail/${stype}/${city}/${district}/${subdistrict}/${id}/${title}-${address}`
  const repKeys: { [key: string]: string } = {
    æ: 'ae',
    å: 'aa',
    ø: 'oe',
    Æ: 'Ae',
    Å: 'Aa',
    Ø: 'Oe',
    á: 'a',
    é: 'e',
    í: 'i',
    ó: 'o',
    ú: 'u',
    Á: 'A',
    É: 'E',
    Í: 'I',
    Ó: 'O',
    Ú: 'U',
    '–': '-',
    '|': '-',
    ä: 'a',
    ë: 'e',
    ï: 'i',
    ö: 'o',
    ü: 'u',
    Ä: 'A',
    Ë: 'E',
    Ï: 'I',
    Ö: 'O',
    Ü: 'U',
  }
  return uri
    .replace(/['()&?,. ]/g, '-')
    .replace(/\*/g, '-')
    .replace(/[æåøÆÅØáéíóúÁÉÍÓÚ–|äëïöüÄËÏÖÜ]/g, char => repKeys[char] || '')
    .replace(/-+/g, '-')
}
