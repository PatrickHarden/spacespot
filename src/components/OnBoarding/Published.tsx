import React from 'react'
import { useIntl } from 'react-intl'
import styled from 'styled-components'
import { useSelector } from 'react-redux'
import { TranslatePropSpace } from 'services/space/types'
import history from 'browserhistory'

import ErrorOutlineRounded from '@material-ui/icons/ErrorOutlineRounded'
import CheckCircleOutlined from '@material-ui/icons/CheckCircleOutlined'
import selectors from 'services/onboarding/selectors'
import CloudImg from 'assets/img/cloud.svg'

import PublishedSpace from './PublishedSpace'
import { SpaceStatus } from 'services/onboarding/types'
import FilledButton from 'components/common/FilledButton'
import Container from './MainContainer'

const Heading = styled.h2`
  text-align: center;
  color: #404042;
  font-size: 25px;
  font-weight: 400;
  line-height: 33px;
  margin: 10px 0;
`

const ErrorMsg = styled.p`
  color: #000000;
  font-size: 16px;
  font-weight: 400;
  line-height: 21px;
  border: 1px solid #db4437;
  border-radius: 2px;
  background-color: rgba(219, 68, 55, 0.3);
  padding: 7px;
`
const IconCloud = styled.div`
  margin-top: 40px;
  text-align: center;
`

const IconOK = styled(CheckCircleOutlined)`
  &.MuiSvgIcon-root {
    font-size: 18px;
    position: relative;
    top: 2px;
    padding-right: 4px;
  }
`

const IconKO = styled(ErrorOutlineRounded)`
  &.MuiSvgIcon-root {
    font-size: 18px;
    position: relative;
    top: 2px;
    padding-right: 4px;
  }
`

const Title = styled.div`
  margin-top: 40px;
  color: black;
  font-size: 18px;
  font-weight: 500;
`
const Line = styled.div<{ width: number; error: boolean }>`
  box-sizing: border-box;
  height: 1px;
  width: 100%;
  border: 2px solid #f2f2f2;
  margin-top: 10px;
  :after {
    display: block;
    position: relative;
    top: -2px;
    content: '';
    box-sizing: border-box;
    height: 1px;
    width: ${props => props.width}%;
    border: 2px solid ${props => (props.error ? '#db4437' : '#4fbbd8')};
  }
`

const ButtonNext = styled(FilledButton)`
  background-color: #404042;
  color: #fffff0;
  float: right;
  margin: 20px 0;
`

const Published = () => {
  const spaces = useSelector(selectors.onboardingState2Space)
  const { formatMessage } = useIntl()
  const t = (s: string) => formatMessage({ id: s })
  const pubSpaces = spaces.filter(
    (space: TranslatePropSpace) =>
      space.status === SpaceStatus.Published ||
      space.status === SpaceStatus.PublishedWithErrors,
  )
  const errorSpaces = spaces.filter(
    (space: TranslatePropSpace) => space.status === SpaceStatus.ErrorPublishing,
  )
  const pendSpaces = spaces.filter(
    (space: TranslatePropSpace) => space.status === SpaceStatus.ReadyForPublish,
  )
  const total = pubSpaces.length + errorSpaces.length + pendSpaces.length
  const pending = pendSpaces.length
  const pct =
    total > 0 ? (100 * (pubSpaces.length + errorSpaces.length)) / total : 100

  return (
    <Container>
      <IconCloud>
        <img alt="publishing" src={CloudImg} />
      </IconCloud>
      <Heading>
        {pending > 0
          ? `${t('ONBOARDING_SAVING')} ${pubSpaces.length + 1} ${t(
              'ONBOARDING_SAVING_OF',
            )} ${total}`
          : t('ONBOARDING_SAVING_END')}
      </Heading>
      <Line width={pct} error={errorSpaces.length > 0} />
      {pubSpaces.length > 0 && (
        <>
          <Title>
            <IconOK htmlColor="#6cb9d5" />
            {t('ONBOARDING_PUBLISHED_OK')}
          </Title>
          {pubSpaces.map((space: TranslatePropSpace) => (
            <PublishedSpace space={space} key={space.key} clickable={false} />
          ))}
        </>
      )}
      {errorSpaces.length > 0 && (
        <>
          <Title>
            <IconKO htmlColor="#db4437" />
            {t('ONBOARDING_PUBLISHED_KO')}
          </Title>
          <ErrorMsg>{t('ONBOARDING_PUBLISH_ERROR_MSG')}</ErrorMsg>
        </>
      )}
      {errorSpaces.map((space: TranslatePropSpace) => (
        <PublishedSpace space={space} key={space.key} clickable={false} />
      ))}
      {pending === 0 && (
        <ButtonNext
          data-testid="publish-ok"
          onClick={() => {
            history.push('/onboarding/validating')
          }}>
          {t('ONBOARDING_CONTINUE')}
        </ButtonNext>
      )}
    </Container>
  )
}
export default Published
