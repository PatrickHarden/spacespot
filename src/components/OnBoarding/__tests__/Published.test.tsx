import React from 'react'
import { mount } from 'enzyme'
import { mountToJson } from 'enzyme-to-json'
import { createOnBoardingState } from '../helpers'

import Published from '../Published'

jest.mock('react-intl', () => ({
  ...jest.requireActual('react-intl'),
  useIntl: () => ({ formatMessage: (s: any) => s.id }),
}))

const mockState = createOnBoardingState()

jest.mock('react-redux', () => ({
  useSelector: () => [mockState],
  useDispatch: () => {
    return jest.fn()
  },
}))

it('renders without crashing ', async () => {
  const wrapper1 = mount(<Published />)
  expect(mountToJson(wrapper1)).toMatchSnapshot()
})
