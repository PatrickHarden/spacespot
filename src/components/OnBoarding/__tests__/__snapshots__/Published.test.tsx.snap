// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing  1`] = `
<Published>
  <styled.div>
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-kpOJdX",
            "isStatic": false,
            "lastClassName": "itQTbV",
            "rules": Array [
              "
  @media (min-width: 1064px) {
    margin: 0 auto;
    padding: 0 20px;
    width: 1024px;
  }
  @media (min-width: 768px) and (max-width: 1063px) {
    margin: 0;
    padding: 0 20px;
  }
  @media (max-width: 767px) {
    margin: 0;
    padding: 0 10px;
  }
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-kpOJdX",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <div
        className="sc-kpOJdX itQTbV"
      >
        <styled.div>
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-jKJlTe",
                  "isStatic": false,
                  "lastClassName": "eCgUwW",
                  "rules": Array [
                    "
  margin-top: 40px;
  text-align: center;
",
                  ],
                },
                "displayName": "styled.div",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-jKJlTe",
                "target": "div",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <div
              className="sc-jKJlTe eCgUwW"
            >
              <img
                alt="publishing"
                src="cloud.svg"
              />
            </div>
          </StyledComponent>
        </styled.div>
        <styled.h2>
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-dxgOiQ",
                  "isStatic": false,
                  "lastClassName": "eUOzyq",
                  "rules": Array [
                    "
  text-align: center;
  color: #404042;
  font-size: 25px;
  font-weight: 400;
  line-height: 33px;
  margin: 10px 0;
",
                  ],
                },
                "displayName": "styled.h2",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-dxgOiQ",
                "target": "h2",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <h2
              className="sc-dxgOiQ eUOzyq"
            >
              ONBOARDING_SAVING_END
            </h2>
          </StyledComponent>
        </styled.h2>
        <styled.div
          error={false}
          width={100}
        >
          <StyledComponent
            error={false}
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-kkGfuU",
                  "isStatic": false,
                  "lastClassName": "fSlNnw",
                  "rules": Array [
                    "
  box-sizing: border-box;
  height: 1px;
  width: 100%;
  border: 2px solid #f2f2f2;
  margin-top: 10px;
  :after {
    display: block;
    position: relative;
    top: -2px;
    content: '';
    box-sizing: border-box;
    height: 1px;
    width: ",
                    [Function],
                    "%;
    border: 2px solid ",
                    [Function],
                    ";
  }
",
                  ],
                },
                "displayName": "styled.div",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-kkGfuU",
                "target": "div",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
            width={100}
          >
            <div
              className="sc-kkGfuU fSlNnw"
              width={100}
            />
          </StyledComponent>
        </styled.div>
        <Styled(styled.button)
          data-testid="publish-ok"
          onClick={[Function]}
        >
          <StyledComponent
            data-testid="publish-ok"
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-iAyFgw",
                  "isStatic": false,
                  "lastClassName": "htVUDI",
                  "rules": Array [
                    "
  font-size: 14px;
  font-weight: 550;
  line-height: 18px;
  border: none;
  background-color: #6cb9d5;
  color: #ffffff;
  border-radius: 2px;
  cursor: pointer;
  text-transform: uppercase;
  height: 37px;
  padding: 0 20px;
",
                    "
  background-color: #404042;
  color: #fffff0;
  float: right;
  margin: 20px 0;
",
                  ],
                },
                "displayName": "Styled(styled.button)",
                "foldedComponentIds": Array [
                  "sc-bwzfXH",
                ],
                "render": [Function],
                "styledComponentId": "sc-iAyFgw",
                "target": "button",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
            onClick={[Function]}
          >
            <button
              className="sc-bwzfXH sc-iAyFgw htVUDI"
              data-testid="publish-ok"
              onClick={[Function]}
            >
              ONBOARDING_CONTINUE
            </button>
          </StyledComponent>
        </Styled(styled.button)>
      </div>
    </StyledComponent>
  </styled.div>
</Published>
`;
